x/mock/types.go

```diff
@@ -0,0 +1,81 @@
+package mock
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/x/auth"
```

Comment:
Remind me again, are we OK with having `x/mock` depend on auth and supply? Will this allow seamless integration?

Reply:
I think the goal is to remove mock after everything has been refactored